site/posts/how-to-commit/index.qmd

```diff
@@ -0,0 +1,164 @@
+
+::: callout-tip
+#### Consolidate all commits of a branch
+
+Even with several commits, you can have a clear view of all the changes when merging to the source branch.
+:::
+
+### Title
+
+The title should be very short (around 50 characters) and explain **what was done** on which file(s) (if relevant).
+
+Starting the title with a verb is very useful to explain what was done, for instance:
+
+::: note
+`Add TRTEMFL in adae.R`
+:::
+
+A naming convention can be defined at sponsor level to identify keyword, such as `Add`, `Fix`, `Remove`, `Update`, etc. The keyword can also be separated form the rest of the title using symbols such as `:` or `!` (for major or breaking changes). For instance:
+
+::: note
+`Add: TRTEMFL in adae.R`
+:::
+
+or
+
+::: note
+`Update! prim endpoint logic in efficacy (adre.R)`
+:::
+
+Using tags that to identify the location of changes can also be an option, such as `SDTM:`, `ADAM:` or `TLF:` for instance.
+
+::: note
+`TLF: create overview of AEs`
+:::
+
+::: callout-tip
+#### Keywords and source code management platforms
+
+So websites such as GitHub are able to recognize some keywords such as `closes` in the title, that will automatically perform an action on the issue or branch (for instance, `closes` will close the issue).
+:::
+
+### Detail
+
+The detail explains **why** the commit needed to be done, the methodological or technical context, or any relevant information that needs to understand the commit.
+
+It should contains, when relevant, the source (such as a new version of the statistical analysis plan, a decision taken in a meeting minutes, an email, etc), the impact (a modification on a ADaM dataset will have an impact on the TL&F describing the updated variables) or the associated logic.
+
+It should be informative but also concise.
+
+Using source code management platforms (GitHub, GitLab, BitBucket, etc) allows a direct link to issues in the title or detail of a commit (for instance `Linked to #35`)
+
+Here is an example of a commit message including a title and the body (detail):
+
+::: note
+**`Add! ANLzzFL in ADxx, ADyy, and ADzz for efficacy`**
+
+`Implements ANLzzFL variable in ADxx, ADyy, and ADzz to identify records for inclusion in the new efficacy analysis introduced in SAP release 3.0 (part 4.5.1, [title]).`
+
+`This flag marks the specific records relevant to the new analysis.`
+
+`Impact: downstream efficacy tables and participant selection for regulatory outputs.`
+:::
+
+::: callout-tip
+#### Conventional commits
+
+[conventionalcommits.org](https://www.conventionalcommits.org/en/v1.0.0/) gives tips and guidance for writing commit messages.
+:::
+
+::: callout-tip
+#### Using AI to improve commits
+Tools such as GitHub Copilot can help to write a good commit message by reviewing a branch and the updated done.
```

Comment:
"and the updated done." 

I think maybe you mean here "by reviewing a branch and what updates were done"?

Reply:
I think this is a good callout (I haven't used this either but i suspect is probably a good use of AI)